presto-main/src/main/java/com/facebook/presto/memory/ClusterMemoryPool.java

```diff
@@ -130,7 +130,7 @@ public synchronized int getAssignedQueries()
 
     public synchronized Map<QueryId, Long> getQueryMemoryReservations()
     {
-        return queryMemoryReservations;
+        return ImmutableMap.copyOf(queryMemoryReservations);
```

Comment:
cmt msg:
> Return a copy in ClusterMemoryPool::getQueryMemoryReservations()

I would call it `Fix thread-safety in ClusterMemoryPool#getQueryMemoryReservations`.
"Return a copy" doesn't convey the problem being fixed

Reply:
Also, move this commit before the code using it, so that there is no point in history with this thread-safety bug. (Just in the unlikely case someone runs git bisect or something)